db/mongodb.go

```diff
@@ -0,0 +1,248 @@
+package db
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/cuducos/minha-receita/transform"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type empresa struct {
+	Cnpj string            `json:"cnpj"`
+	Json transform.Company `json:"json"`
+}
+
+type MongoDB struct {
+	client *mongo.Client
+	db     *mongo.Database
+	ctx    context.Context
+}
+
+// NewMongoDB initializes a new MongoDB connection wrapped in a structure.
+func NewMongoDB(uri string) (MongoDB, error) {
+	opts := options.Client().ApplyURI(uri)
+	ctx := context.Background()
+	c, err := mongo.Connect(ctx, opts)
+	if err != nil {
+		return MongoDB{}, fmt.Errorf("failed to connect to MongoDB: %w", err)
+	}
+	if err := c.Ping(ctx, nil); err != nil {
+		return MongoDB{}, fmt.Errorf("failed to ping to MongoDB: %w", err)
+	}
+	u := strings.Split(uri, "?")[0] // Remove query parameters from the URI
+	ps := strings.Split(u, "/")
+	dbName := ps[len(ps)-1]
+
+	// Ensure the extracted database name is valid
+	if dbName == "" || strings.Contains(dbName, "@") {
+		return MongoDB{}, fmt.Errorf("No database name found in the URI")
```

Comment:
Corrigindo o erro que a CI acusa

```suggestion
		return MongoDB{}, fmt.Errorf("no database name found in the URI")
```